integrations/argocd/client.py

```diff
@@ -66,6 +75,7 @@ async def get_application_by_name(self, name: str) -> dict[str, Any]:
 
     async def get_deployment_history(self) -> list[dict[str, Any]]:
         """The ArgoCD application route returns a history of all deployments. This function reuses the output of the application endpoint"""
+        logger.warning(DEPRECATION_WARNING)
```

Comment:
there is a dedicated package to log deprecation notices 
https://pypi.org/project/deprecation/